ApiReview/API.Polly.Core/NoDocs/Polly/ResilienceContext.cs

```diff
@@ -0,0 +1,20 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Polly.Telemetry;
+
+namespace Polly;
+
+public sealed class ResilienceContext
+{
+    public string? OperationKey { get; }
+    public CancellationToken CancellationToken { get; }
+    public bool ContinueOnCapturedContext { get; }
+    public ResilienceProperties Properties { get; }
+    public IReadOnlyList<ResilienceEvent> ResilienceEvents { get; }
```

Comment:
I have mixed feelings about `ResilienceEvents` property - on one side, it looks like a nice addition that **some** might benefit from, on the other given the current implementation of the strategies and that the list can grow large in case of multiple retries, hedging, etc - we probably should make the logic that records such events (1) optional and (2) customizable.

I suggest to revisit this.

Reply:
The important information is not individual events itself rather the information on whether the execution was degraded or not. So I think we can indeed drop this list and replace is with something like `IsExecutionHealthy` or `IsExecutionDegraded` flag. Essentially this flag becomes true whener resilience strategy reports event with severity greater than `Information`.

If someone needs this we can put it back.

Reply:
Hide it for V8 initial release.

Reply:
Just drop it in initial release and wait for feedback.